go/tao/tpm2_tao.go

```diff
@@ -285,47 +291,100 @@ func NewTPM2Tao(tpmPath string, statePath string, pcrNums []int) (Tao, error) {
 	tt.pcrs = pcrNums
 	tt.path = statePath
 
-	// Create the root key.
 	keySize := uint16(2048)
 	quotePassword := ""
 
-	rootSaveContext := path.Join(tt.path, "root_context")
-	_, rootErr := os.Stat(rootSaveContext)
+	quoteKeyPrivateBlobFile := path.Join(tt.path, "quote_private_key_blob")
+	_, quotePrivateErr := os.Stat(quoteKeyPrivateBlobFile)
+	quoteKeyPublicBlobFile := path.Join(tt.path, "quote_public_key_blob")
+	_, quotePublicErr := os.Stat(quoteKeyPrivateBlobFile)
 
-	quoteSaveContext := path.Join(tt.path, "quote_context")
-	_, quoteErr := os.Stat(quoteSaveContext)
+	sealKeyPrivateBlobFile := path.Join(tt.path, "seal_private_key_blob")
+	_, sealPrivateErr := os.Stat(sealKeyPrivateBlobFile)
+	sealKeyPublicBlobFile := path.Join(tt.path, "seal_public_key_blob")
+	_, sealPublicErr := os.Stat(sealKeyPrivateBlobFile)
 
-	sealSaveContext := path.Join(tt.path, "seal_context")
-	_, sealErr := os.Stat(sealSaveContext)
+	var quoteKeyPublicBlob []byte
+	var quoteKeyPrivateBlob []byte
+	var sealKeyPublicBlob []byte
+	var sealKeyPrivateBlob []byte
 
-	if rootErr != nil || quoteErr != nil || sealErr != nil {
-		if err := tpm2.InitTpm2Keys(tt.rw, tt.pcrs, keySize, uint16(tpm2.AlgTPM_ALG_SHA1), quotePassword, rootSaveContext, quoteSaveContext, sealSaveContext); err != nil {
+	// Create the root key.
+	rootHandle, err := tpm2.CreateTpm2HierarchyRoot(tt.rw, tt.pcrs, keySize, uint16(tpm2.AlgTPM_ALG_SHA1))
+	if err != nil {
+		return nil, err
+	}
+	defer tpm2.FlushContext(tt.rw, rootHandle)
+
+	if err != nil || quotePrivateErr != nil || sealPrivateErr != nil || quotePublicErr != nil || sealPublicErr != nil {
```

Comment:
what if some files exists and some files don't? Is that just considered a bad state, and so we regenerate all keys?
also, minor, but err != nil doesn't do anything, since we return nil on lines 314-316 if something went wrong there.